src/java.base/share/classes/java/text/DigitList.java

```diff
@@ -153,7 +153,8 @@ public void clear () {
      */
     public void append(char digit) {
         if (count == digits.length) {
-            char[] data = new char[count + 100];
+            char[] data = new char[count > Integer.MAX_VALUE / 2 ?
+                    Integer.MAX_VALUE : count * 2];
```

Comment:
I think you can use `ArraysSupport.newLength(count, 1, count)` instead.

Reply:
Did not know we had that, thanks!